synapse/handlers/federation_event.py

```diff
@@ -1315,6 +1320,42 @@ async def _handle_marker_event(self, origin: str, marker_event: EventBase) -> No
             marker_event,
         )
 
+    async def backfill_event(
+        self, destination: str, room_id: str, event_id: str
+    ) -> None:
+        room_version = await self._store.get_room_version(room_id)
+
+        logger.info("backfill_event event_id=%s", event_id)
+
+        eventAsdf = await self._federation_client.get_pdu(
+            [destination],
+            event_id,
+            room_version,
+        )
+        event = await self._federation_client.get_pdu_from_destination_raw(
+            destination,
+            event_id,
+            room_version,
+        )
```

Comment:
What makes the event returned by `get_pdu_from_destination_raw(...)` work when processed by `_process_pulled_events` but not `get_pdu(...)`? By work, I mean is accepted and the test in https://github.com/matrix-org/complement/pull/406 passes

I know that `get_pdu_from_destination_raw(...)` returns a non-outlier but there is an additional difference because setting `event.internal_metadata.outlier = False` for `get_pdu(...)` still doesn't work completely.

Reply:
The difference is just the `outlier` vs non-`outlier`. We want a non-`outlier` so that `_process_pulled_events()` -> `_update_outliers_txn()` creates a `state_group` and mimics what would happen in a regular backfill. `get_pdu()` can potentially return an `outlier` depending on the cache which we don't want.

Things also didn't work before because I was directly editing the `event.internal_metadata.outlier` reference which is also referenced in other places like the cache. So I was unintentionally change the cache source of truth which threw other things off.